packages/ibm-products/src/components/Decorator/Decorator.js

```diff
@@ -0,0 +1,354 @@
+/**
+ * Copyright IBM Corp. 2024, 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useMemo } from 'react';
+
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import { getDevtoolsProps } from '../../global/js/utils/devtools';
+import { pkg } from '../../settings';
+
+import { getMagnitude, truncate } from './constants';
+import { DecoratorIcon } from './DecoratorIcon';
+
+const blockClass = `${pkg.prefix}--decorator`;
+const componentName = 'Decorator';
+
+const defaults = {
+  labelTitle: (score, scoreThresholds, magnitude) => {
+    if (typeof score !== 'number') {
+      return 'Unknown score';
+    }
+    return `"${magnitude}" magnitude. Score ${score} out of ${
+      scoreThresholds.slice(-1)[0]
+    }`;
+  },
+  scoreThresholds: [0, 4, 7, 10],
+  theme: 'light',
+};
+
+/**
+ * The Decorator groups a key/value pair to look and behave like a single UI element.
+ */
+export let Decorator = React.forwardRef(
+  (
+    {
+      // The component props, in alphabetical order (for consistency).
+      className,
+      disabled,
+      hideIcon,
+      href,
+      label,
+      labelTitle = defaults.labelTitle,
+      onClick,
+      onClickLabel,
+      onClickValue,
+      onContextMenu,
+      score,
+      scoreThresholds = defaults.scoreThresholds,
+      small,
+      theme = defaults.theme,
+      truncateValue,
+      value,
+      valueTitle,
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    const magnitude = useMemo(() => {
+      return getMagnitude(score, scoreThresholds);
+    }, [score, scoreThresholds]);
+
+    const _labelTitle = useMemo(() => {
+      return labelTitle && labelTitle(score, scoreThresholds, magnitude);
+    }, [magnitude, score, labelTitle, scoreThresholds]);
+
+    // If a "midline" truncation has been defined,
+    // then use the formatted midline value generated here,
+    // else refer to the original value.
+    const _value = useMemo(() => {
+      if (truncateValue?.maxLength) {
+        return truncate(
```

Comment:
you can make this parameter signature cleaner by just passing in `(value, truncateValue)` and then destructing the values you need in `truncate`